bpf/lib/overloadable_skb.h

```diff
@@ -73,6 +73,11 @@ redirect_self(const struct __sk_buff *ctx)
 	 */
 #ifdef ENABLE_HOST_REDIRECT
 	return redirect(ctx->ifindex, 0);
+#elif NATIVE_IFINDEX
+	if (ctx->ifindex == NATIVE_IFINDEX)
+		return redirect(ctx->ifindex, 0);
+	else
+		return redirect(ctx->ifindex, BPF_F_INGRESS);
```

Comment:
trying to keep the old behavior as much as possible. I am worried to break old logic.

Reply:
I mean why is any change to this file required? You are not even calling `redirect_self()`, right?

Reply:
if does call `redirect_self()`: `send_icmp6_ndisc_adv => icmp6_send_reply => redirect_self`

Reply:
I still don't understand why this change is needed.
1. I couldn't find any definition of `NATIVE_IFINDEX` in the source code. Where is it defined? Should it be `NATIVE_DEV_IFINDEX`?
2. Assuming `NATIVE_IFINDEX` exists and is the index of a native device, why do we need to redirect packets to the _ingress_ of that device?

Reply:
Originally it was only doing `redirect(ctx->ifindex, BPF_F_INGRESS);`, actually I don't understand which code path will hit this. I am guessing it could be `cilium_host` and redirect the packet back to the pods. Now the patch basically added a direction. If the packet is coming from native interface, redirect as egress, otherwise still ingress.

Reply:
This is not useful after rebasing on master. I've removed it.